src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -399,6 +469,22 @@ Status CoreWorkerDirectTaskSubmitter::CancelTask(TaskSpecification task_spec,
             cancel_retry_timer_->async_wait(boost::bind(
                 &CoreWorkerDirectTaskSubmitter::CancelTask, this, task_spec, force_kill));
           }
+        } else if (status.ok() && reply.attempt_succeeded()) {
```

Comment:
I don't think this is necessary because we should still get back the callback for `PushNormalTask`.